### PR TITLE
Add emoji support

### PR DESCRIPTION
**Pull request by calve**
Add a new `emoji` theme, using emojis as file icons. Icons are set
depending of the file extension

Rendering depends of the font used and may differ between users.
Most emojis width are twice as a single ASCII character, it's up to the
terminal application to render them accordingly.

Example display :

```
📁..                |
📂icons/            |
  📄close.xpm       |
  📄leaf.xpm        |
  📄open.xpm        |
📂screenshots/      |
  🌄icons.png       |
📂test/             |
  🐏neotree-test.el |
  🐏test-buffer.el  |
  🐏test-cmds.el    |
  🐏test-utils.el   |
  🐏test-vc.el      |
📄LICENSE           |
🔧Makefile          |
📘README.md         |
🐏neotree.el        |
```

### CODE REVIEW

**Comment by calve** [comment]
Hi ! This is my very first emacs-lisp contribution, so maybe some code can be simplified ?

File: neotree.el [hunk 1]
```diff
@@ -1237,12 +1239,70 @@ Return nil if DIR is not an existing directory."
...
+   (".jpeg" . "🌄")
+   (".jpg" . "🌄")
+   (".key" . "🔒")
+   (".md" . "📘")
+   (".mp3" . "🔈")
+   (".ocaml" . "🐪")
+   (".pdf" . "📰")
+   (".php" . "🐘")
+   (".png" . "🌄")
+   (".pubkey" . "🔑")
```
Comment (milouse) [inline neotree.el:1261]: You should use the same emoji for the `.asc` extension.

File: neotree.el [hunk 2]
```diff
@@ -1237,12 +1239,70 @@ Return nil if DIR is not an existing directory."
...
+   (".py" . "🐍")
+   (".rb" . "🔺")
+   (".rst" . "📗")
+   (".sh" . "🐚")
+   (".sls" . "🧂")
+   (".text" . "📝")
+   (".txt" . "📝")
```
Comment (milouse) [inline neotree.el:1268]: Don’t forget the `.org` extension ;)

File: neotree.el [hunk 3]
```diff
@@ -1237,12 +1239,70 @@ Return nil if DIR is not an existing directory."
...
+   (".yaml" . "🔧")
+   (".yml" . "🔧")
+   (".zsh" . "🐚")
+   ("Dockerfile" . "🐋")
+   ("LICENSE" . "📕")
+   ("Makefile" . "🔧")
+   ("README" . "📕")
+   ("Vagrantfile" . "💻")
+   )
+  )
```
Comment (milouse) [inline neotree.el:1280]: you should not leave floating ending parenthesis and just pad all of them at the last line of content. i.e. `("Vagrantfile" . "🖥️")))`

File: neotree.el [hunk 4]
```diff
@@ -1237,12 +1239,70 @@ Return nil if DIR is not an existing directory."
...
+
+(defun neo-emoji-get-extension (filename)
+  "Return the extension of 'FILENAME'.
+The extension is defined as the substring after the last dot, last dot included.
+If no extension can be found, return the whole 'FILENAME'."
+  (let ((extension (file-name-extension filename t)))
+    (if (string-blank-p extension) filename extension))
+  )
+
+(defun neo-emoji-icon-for-dir (dirname direction)
```
Comment (milouse) [inline neotree.el:1290]: As you are not using it, I think you should remove the `dirname` argument for this method and just keep `direction` alone.

Thus you can simplify this method call on lines 1324 and 1325.

**Review by milouse — COMMENTED** [review]
Hi, I just found your MR, which is a great idea :)

Here are my little two cents to your work.

**Comment by milouse** [comment]
@Abdisalan can I ask why this MR has been closed? Do you reject the feature or is it just because the MR is very old and this branch would need a rebase? If the later, please let me know, I’d provide such a rebase on last dev version.

**Comment by Abdisalan** [comment]
@milouse I closed it because this feature is already covered by the `all-the-icons` package integration. If you want icons in the terminal, I think a better solution would be to have _something like_ `all-the-icons` so that the icons match the file extension better and is more comprehensive.

This PR only supports ~30 extensions and a lot of the emojis either don't really match the extension or have a duplicate wrench which makes it confusing.

**Pull request closed without merging**